sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -253,7 +259,7 @@ class FileScanRDD(
                     null
                   // Throw FileNotFoundException even if `ignoreCorruptFiles` is true
                   case e: FileNotFoundException if !ignoreMissingFiles => throw e
-                  case e @ (_: RuntimeException | _: IOException) if ignoreCorruptFiles =>
+                  case e @ (_: RuntimeException | _: IOException) if shouldSkipCorruptFiles() =>
```

Comment:
`HadoopRDD` and `NewHadoopRDD` also use `ignoreCorruptFiles` although they are not sql apis, don't they need similar change?